src/providers/UrqlProvider.tsx

```diff
@@ -0,0 +1,46 @@
+import { createClient, Provider, Client } from 'urql'
+import { Logger } from '@oceanprotocol/lib'
+import React, { useState, useEffect, ReactNode, ReactElement } from 'react'
+import { useWeb3 } from './Web3'
+import { getOceanConfig } from '../utils/ocean'
+
+let urqlClient: Client
+
+function createUrqlClient(subgraphUri: string) {
+  const client = createClient({
+    url: `${subgraphUri}/subgraphs/name/oceanprotocol/ocean-subgraph`
+  })
+  return client
+}
+
+export function getUrqlClientInstance(): Client {
+  return urqlClient
+}
+
+export default function UrqlClientProvider({
+  children
+}: {
+  children: ReactNode
+}): ReactElement {
+  const { networkId } = useWeb3()
+  const [client, setClient] = useState<Client>()
+
+  useEffect(() => {
+    const oceanConfig = getOceanConfig(networkId || 1)
+
+    if (!oceanConfig?.subgraphUri) {
+      Logger.error(
+        'No subgraphUri defined, preventing UrqlProvider from initialization.'
+      )
+      return
+    }
+
+    const newClient = createUrqlClient(oceanConfig.subgraphUri)
+    urqlClient = newClient
+    setClient(newClient)
+    Logger.log(`[apollo] Client connected to ${oceanConfig.subgraphUri}`)
+  }, [networkId])
```

Comment:
Switching the `subgraphUri` based on `networkId` is no longer because we can have assets from different networks than the connected wallet.  When retrieving data for an asset It could be triggered by `ddo` change, but in other cases we might need to hit multiple subgraphs. So I don't know how reliable is to keep a client in provider, might be better to create a function which accepts a list of subgraphUri's and create the client when calling the function.

Reply:
yep this will be removed, it was left here just for testing reasons for the urql library with one of our subgraphs

Reply:
the question how to get the `subgraphUri`s we need is also open question still, as getting them one-by-one with `getOceanConfig()` might be not really feasible anymore. For that reason ocean.js now exports the whole network metadata definition at once in case we need to iterate over that data: https://github.com/oceanprotocol/ocean.js/blob/main/src/utils/ConfigHelper.ts#L34

As for keeping the client in some provider here, we need to test to see. It depends on how quick the client can be initiated as to optimize for the most common user flow which is a user browsing assets on mainnets. So maybe makes sense keeping some connection in some sort of cache so browsing feels faster for user. But maybe it doesn't really matter as connection is quick anyway, then as Norbert suggests the utility method could be enough